rcl/include/rcl/client.h

```diff
@@ -394,7 +394,7 @@ rcl_client_get_rmw_handle(const rcl_client_t * client);
  */
 RCL_PUBLIC
 bool
-rcl_client_is_valid(const rcl_client_t * client);
+rcl_client_is_valid(const rcl_client_t * client, const rcl_allocator_t * error_msg_allocator);
```

Comment:
Why should the passed allocators be `const`?

Same for all signatures.

Reply:
My guess is that it's because that's what is used for the `node_is_valid` and that's just been propagated to those. I don't feel strongly about it though

Reply:
Since the [internally called function](https://github.com/ros2/rcutils/blob/9fc616d389372d3bec080e8965e23f4acac7f29f/include/rcutils/error_handling.h#L101-L102) takes a non-const allocator this should neither be const.

Reply:
sounds good https://github.com/ros2/rcl/pull/180/commits/ec3a66f0ce62c6832575d55f9d9a0ab053bdd022